pokemon/src/components/DetailPage/index.tsx

```diff
@@ -0,0 +1,77 @@
+import clsx from 'clsx';
+import { GRADIENT_VARIANTS } from './constants';
+import AbilitySvg from '@assets/information-icons/ability.svg';
+import HeightSvg from '@assets/information-icons/height.svg';
+import WeightSvg from '@assets/information-icons/weight.svg';
+import SpeciesSvg from '@assets/information-icons/species.svg';
+import Spinner from '@components/Spinner';
+import StatChip from '@components/StatChip';
+import TypeChip from '@components/TypeChip';
+import TypeSvg from '@components/TypeSvg';
+import { PkmnNumberFormat } from '@helpers/formatters';
+import { usePokemon } from '@contexts/Pokemon';
+
+const DetailPage = () => {
+  const { selected, isLoadingDetails } = usePokemon();
+
+  if (!selected) {
+    return null;
+  }
+
+  const {
+    types,
+    image: pkmnSprite,
+    name,
+    id,
+    weight,
+    ability,
+    species,
+    height,
+  } = selected;
+
+  const [mainType, secondaryType] = types;
+
+  const headerClassName = clsx(
+    'relative h-[307px] bg-gradient-to-br',
+    `${GRADIENT_VARIANTS[mainType]}`
+  );
+
+  const formattedPkmnNo = PkmnNumberFormat.format(id);
+
+  return (
+    <section className="relative h-full">
+      {isLoadingDetails && <Spinner />}
```

Comment:
NOTE: going from blank -> pokemon does not show loading state

Reply:
I was about to write how I'm wondering how the loading state works in that case, but I think it's just an oversight on my part from not rendering the details page if a Pokémon is not selected a few lines above this hahaha